### ISSUE TITLE
Support subscriptions

### ISSUE THREAD

**Issue by xaviershay** [caraus-ecms/graphql#5]
I would expect this to parse:

```
*Main Data.Attoparsec.Text Data.GraphQL.Parser Data.Text> parseOnly document (Data.Text.pack "query HeroName { hero { name } }")
Right (Document [DefinitionOperation (Query (Node "HeroName" [] [] [SelectionField (Field "" "hero" [] [] [SelectionField (Field "" "name" [] [] [])])]))])
*Main Data.Attoparsec.Text Data.GraphQL.Parser Data.Text> parseOnly document (Data.Text.pack "subscription HeroName { hero { name } }")
Left "document error!: string"
```

I can't find any reference in code to subscription, so I figure it's just a thing in the TODO list!

Investigation notes

* Can probably start by adding as a case to everywhere `Mutation` shows in the code: https://github.com/caraus-ecms/graphql/search?q=mutation&unscoped_q=mutation
* I don't understand the `initialValue` discussion here yet, but probably important: https://github.com/haskell-graphql/graphql-api/issues/209
* Also some relevant thoughts here https://github.com/haskell-graphql/graphql-api/issues/203

I have no idea what I'm doing here, as part of learning graphql I thought it would be fun to implement a small subscriptions thing ... and I've fallen into the rabbit hole :) Might poke further on this issue later in the week.

**Comment by belka-ew** [comment]
Subscirptions are indeed not implemented at all. Adding parser support should be trivial, but I'm not sure, what else has to be done, since I haven't used GraphQL subscriptions yet.

This library was unmaintained for several years, I recently took over the maintenance and I'm permanently working on it, but in baby steps. And there are other important things that need love and care :), like schema validation, which is a big and important issue as well. So I'll be honest, the implementation of the full GraphQL specification will take some time (or help from contributors). That said I'm already using the library for simple queries and mutations and it works fine.

Anyway if you use the library and have questions or encounter bugs, please open issues, it would help me to improve the documentation and fix possible problems.

Looking at your ghci output you seem to use an older version, I replaced Attoparsec with Megaparsec (to improve error messages) and moved all the code under `Language.GraphQL` (instead of `Data.GraphQL`) since `Language.GraphQL` seems more appropriate to me. I've released these changes yesterday under 0.4.0.0.

**Comment by belka-ew** [comment]
Done.

The implementation is based on conduit and is pretty boring. There is a new resolver data constructor that should create a source event stream. The executor receives the events, pipes them through the normal execution and puts them into the response stream which is returned to the user. Conduit takes the most work with the streams.

There is one test, that shows the usage. I still need to update the "high-level" functions, that combine different GraphQL stages into a single pipeline. That will be done soon.
